File: MyApi/Models/Comanda.cs
using MyApi.Enums;

namespace MyApi.Models
{
    public class Comanda
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public User User { get; set; } = null!;

        public string Strada { get; set; } = string.Empty;
        public string Oras { get; set; } = string.Empty;
        public string Judet { get; set; } = string.Empty;
        public string Tara { get; set; } = string.Empty;
        public string CodPostal { get; set; } = string.Empty;
        public string NumeDestinatar { get; set; } = string.Empty;
        public string TelefonDestinatar { get; set; } = string.Empty;

        public DateTime DataCrearii { get; set; }
        public StatusComanda Status { get; set; } = StatusComanda.Noua;
        public decimal Total { get; set; }

        public ICollection<ElementComanda> ElementeComanda { get; set; } = [];

        public byte[] RowVersion { get; set; } = [];
    }
}

File: MyFrontend/src/types/Comanda.ts
export interface ElementComandaCreate {
  produsId: number
  cantitate: number
}

export interface ComandaCreateRequest {
  adresaId: number
  elementeComandaCreateDto: ElementComandaCreate[]
}

export interface ElementComanda {
  id: number
  produsId: number
  numeProdus: string
  cantitate: number
  pretUnitar: number
}

export interface Comanda {
  id: number
  userId: number

  strada: string
  oras: string
  judet: string
  tara: string
  codPostal: string
  numeDestinatar: string
  telefonDestinatar: string

  dataCrearii: string
  status: string
  total: number

  elementeComanda: ElementComanda[]
  rowVersion: string
}